compiler/compile/OMRCompilation.cpp

```diff
@@ -1839,9 +1839,11 @@ void OMR::Compilation::resetVisitCounts(vcount_t count, TR::TreeTop *start)
 
 void OMR::Compilation::reportFailure(const char *reason)
    {
-   traceMsg(self(), "Compilation Failed Because: %s\n", reason);
    if (self()->getOption(TR_PrintErrorInfoOnCompFailure))
+      {
+      traceMsg(self(), "Compilation Failed Because: %s\n", reason);
```

Comment:
The traceMsg shouldn't be hidden under a condition - if the compile fails the reason should always be shown in any trace log attached to the compile.